FilePersistence/src/version_control/merge/ChangeGraph.cpp

```diff
@@ -448,14 +446,105 @@ void ChangeGraph::removeLabelConflictsBetweenChildren(Model::NodeIdType parentId
 	}
 }
 
-void ChangeGraph::updateIncomingLabels(Model::NodeIdType /*parentId*/, IdToLabelMap /*labelMap*/)
+void ChangeGraph::updateListChanges(Model::NodeIdType parentId, IdToLabelMap labelMap, GenericTree* baseTree)
 {
-	Q_ASSERT(false);
-}
+	for (auto nodeId : labelMap.keys())
+	{
+		QString labelA;
+		QString labelB;
+		QString labelNone;
+		auto labels = labelMap.values(nodeId);
+		for (auto label : labels)
+		{
+			if (label.branch_.testFlag(MergeChange::BranchA))	labelA = label.label_;
+			else if (label.branch_.testFlag(MergeChange::BranchB))	labelB = label.label_;
+			else labelNone = label.label_;	//Base
+		}
 
-void ChangeGraph::updateOutgoingLabels(Model::NodeIdType /*parentId*/, IdToLabelMap /*labelMap*/)
-{
-	Q_ASSERT(false);
+		// Need to adjust Update Changes as well
+		auto parentNode = baseTree->find(parentId);
```

Comment:
You can bring this and the qassert below outside of the main loop. It won't change.